### PR TITLE
fix(DB): Missing special dialogue for High Chief Stillpine during 'The Kessel Run'

### PR DESCRIPTION
**Pull request by SuspiciousTucan**
During the quest 'The Kessel Run' (id: 9663) High Chief Stillpine (id:
17440) wouldn't show the correct dialogue, displaying the default one instead.
This was due to a missmatch between the creature_template.gossip_menu_id and gossip_menu.MenuID.

## Changes Proposed:
-  Update quest dialogue's gossip_menu to point to the correct gossip_menu_id. 

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/6883
- Closes https://github.com/chromiecraft/chromiecraft/issues/1158

## SOURCE:

- [youtube video of the quest on chromiecraft](https://www.youtube.com/watch?v=OfoXTy8wlaw)
- [source for correct dialogue](https://wowpedia.fandom.com/wiki/The_Kessel_Run)

## Tests Performed:

- Tested on Windows 10, no modules.

## How to Test the Changes:

1. .go c 17440
2. Interacting with 17440 will provide default gossip.
3. .q add 9663
4. Interacting with 17440 now provide a quest specific gossip

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1626260907562417400.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1626260907562417400');
+UPDATE gossip_menu SET MenuID = 7434 WHERE MenuID = 7983 AND TextID = 9039;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1626260907562417400.sql:2]:
```suggestion
UPDATE `gossip_menu` SET `MenuID` = 7434 WHERE `MenuID` = 7983 AND `TextID` = 9039;
```
https://www.azerothcore.org/wiki/sql-standards#general-standards

**Review by Kitzunu — CHANGES_REQUESTED** [review]

**Comment by FrancescoBorzi** [comment]
hello @SuspiciousTucan and welcome to AzerothCore!

thanks for sending this PR, however it's failing with the following:

```
addToGossipMenus: GossipMenu 7983 not found
Not handled grouped condition, SourceGroup 7983
```

can you please fix that? you can see the failure here on GitHub, just click on "Details" on any of the failing builds:

image

**Comment by Helias** [comment]
@SuspiciousTucan any updates?

**Comment by SuspiciousTucan** [comment]
Evening,

sorry I had other deadlines in the previous weeks. I'll take a look at it on sunday and let you know.

**Comment by Helias** [comment]
no worries at all, just to know, thanks!

**Comment by Asthalor** [comment]
Just to add some info:

Happened to me when launching the program.
It looks like there is a condition that added some gossip, so its very likely that there were some conditions that used that menu id.

image

The SQL that link to that gossip is this one

SELECT * FROM conditions WHERE SourceTypeOrReferenceId = 14 AND sourcegroup = 7983;

image

I hope that helps.

**Comment by Helias** [comment]
Thanks @Asthalor, I imported your fix in the PR.

**Comment by Helias** [comment]
the issue still persists

**Comment by Asthalor** [comment]
I think the delete query i passed was after doing the update the author does on the PR, so we still have the `SourceGroup` = 7983. My bad on that!

I think we need to do

DELETE FROM `conditions` WHERE (`SourceTypeOrReferenceId` = 14) AND (`SourceGroup` = 7983) AND (`SourceEntry` = 9039) AND (`SourceId` = 0) AND (`ElseGroup` = 0) AND (`ConditionTypeOrReference` = 9) AND (`ConditionTarget` = 0) AND (`ConditionValue1` = 9663) AND (`ConditionValue2` = 0) AND (`ConditionValue3` = 0);

So we delete the bad condition. And then we add that again with 

DELETE FROM `conditions` WHERE (`SourceTypeOrReferenceId` = 14) AND (`SourceGroup` = 7434) AND (`SourceEntry` = 9039) AND (`SourceId` = 0) AND (`ElseGroup` = 0) AND (`ConditionTypeOrReference` = 9) AND (`ConditionTarget` = 0) AND (`ConditionValue1` = 9663) AND (`ConditionValue2` = 0) AND (`ConditionValue3` = 0);

INSERT INTO `conditions` (`SourceTypeOrReferenceId`, `SourceGroup`, `SourceEntry`, `SourceId`, `ElseGroup`, `ConditionTypeOrReference`, `ConditionTarget`, `ConditionValue1`, `ConditionValue2`, `ConditionValue3`, `NegativeCondition`, `ErrorType`, `ErrorTextId`, `ScriptName`, `Comment`) VALUES
(14, 7434, 9039, 0, 0, 9, 0, 9663, 0, 0, 0, 0, 0, '', 'Show gossip text 9039 if player has quest 9663');

After that we can do the PR commit 

UPDATE `gossip_menu` SET `MenuID` = 7434 WHERE `MenuID` = 7983 AND `TextID` = 9039;

**Review by FrancescoBorzi — APPROVED** [review]

**Pull request merged**